DokuwikiXMLExport.php

```diff
@@ -0,0 +1,194 @@
+            return false;
+        }
+
+        $items = array();
+        for ($i = $start; $i < $start + $count; $i++) {
+            $item = $exporter->createItem($i);
+
+            $name = new Name();
+            $name->setValue($this->getName($pages, $i));
+            $item->setName($name);
+
+            $summary = new Summary();
+            $summary->setValue($this->getSummaryAndDescription($pages, $i));
+            $item->setSummary($summary);
+
+            $description = new Description();
+            $description->setValue($this->getSummaryAndDescription($pages, $i));
+            $item->setDescription($description);
+
+            $price = new Price();
+            $price->setValue('0.0');
+            $item->setPrice($price);
+
+            $Url = new Url();
+            $Url->setValue($this->getUrl($pages, $i));
+            $item->setUrl($Url);
+
+            $dateAdded = new DateAdded();
+            $dateAdded->setDateValue($this->getDateAdded($pages, $i));
+            $item->setDateAdded($dateAdded);
+
+            $item->addOrdernumber(new Ordernumber($this->getOrdernumber($pages, $i)));
+            $items[] = $item;
+
+            $attributeCategory = new Attribute('cat', $this->getAttributesCategory($pages, $i));
+            $item->addAttribute($attributeCategory);
+        }
+        return $exporter->serializeItems($items, $start, $total);
+    }
+
+
+    /**
+     * Gets the Ordernumber of the current page.
+     *
+     * @param array $pages Contains all namespaces of all DokuWiki pages.
+     * @param integer $key The Item ID.
+     * @return string Returns the ordernumber.
+     */
+    private function getOrdernumber($pages, $key)
+    {
+        return $pages[$key];
+    }
+
+    /**
+     * Gets the Name of the current page.
+     *
+     * @param array $pages Contains all namespaces of all DokuWiki pages.
+     * @param integer $key The Item ID.
+     * @return string Returns the Name/Title of the page.
+     */
+    private function getName($pages, $key)
+    {
+        $metadata = p_get_metadata($pages[$key]);
+        return $metadata["title"];
+    }
+
+    /**
+     * Gets the Summary and Description of the current page.
+     *
+     * @param array $pages Contains all namespaces of all DokuWiki pages.
+     * @param integer $key The Item ID.
+     * @return string Returns the Summary and Description of the page.
+     */
+    private function getSummaryAndDescription($pages, $key)
+    {
+        $metadata = p_get_metadata($pages[$key]);
+        return $metadata["description"]["abstract"];
+    }
+
+    /**
+     * Gets the Url of the current page.
+     *
+     * @param array $pages Contains all namespaces of all DokuWiki pages.
+     * @param integer $key The Item ID.
+     * @return string Returns the Url of the page.
+     */
+    private function getUrl($pages, $key)
+    {
+        $url = DOKU_URL . 'doku.php?id=' . $this->getOrdernumber($pages, $key);
```

Comment:
Find a Dokuwiki method that generated the full URL including `doku.php?id=` for you or replace with constant(s).

Reply:
TODO:

- Search for a method that creates an URL according to parameters that I give this function.